src/item/HorseArmor.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\item;
+
+use pocketmine\color\Color;
+use pocketmine\nbt\tag\CompoundTag;
+use pocketmine\nbt\tag\IntTag;
+use pocketmine\utils\Binary;
+
+class HorseArmor extends Item{
+
+	public const TAG_CUSTOM_COLOR = "customColor"; //TAG_Int
+
+	protected ?Color $customColor = null;
+
+	/**
+	 * @param string[] $enchantmentTags
+	 */
+	public function __construct(ItemIdentifier $identifier, string $name, array $enchantmentTags = []){
+		parent::__construct($identifier, $name, $enchantmentTags);
+	}
+
+	public function getMaxStackSize() : int{
+		return 1;
+	}
+
+	/**
+	 * Returns the dyed colour of this armour piece. This generally only applies to leather armour.
+	 */
+	public function getCustomColor() : ?Color{
+		return $this->customColor;
+	}
+
+	/**
+	 * Sets the dyed colour of this armour piece. This generally only applies to leather armour.
+	 *
+	 * @return $this
+	 */
+	public function setCustomColor(Color $color) : self{
+		$this->customColor = $color;
+		return $this;
+	}
+
+	/** @return $this */
+	public function clearCustomColor() : self{
+		$this->customColor = null;
+		return $this;
+	}
```

Comment:
Permit null in the setter instead of just creating a new function for that

Reply:
This code is by Armor.php but I can replace it if you want

Reply:
I think clearCustomColor() is needed if you want to remove the color via plugin, to me clearCustomColor() sounds more logical than setCustomColor(null), doesn't it?

Reply:
Oops then someone screwed up badly at Armor.php if you have this opinion.

But I'm happy to change it for you, but as I said, I think it's unnecessary

Reply:
Its just old code, the reflexion is not new, but to remove all the clear* shit method we need to break BC

Reply:
jo all good has already been adjusted